src/DataStructures/SpinWeighted.hpp

```diff
@@ -69,6 +70,48 @@ struct SpinWeighted<T, Spin, false> {
     return *this;
   }
 
+  template <typename Rhs>
+  auto& operator+=(const SpinWeighted<Rhs, Spin>& rhs) noexcept {
+    data_ += rhs.data();
+    return *this;
+  }
+
+  template <typename Rhs>
+  auto& operator+=(SpinWeighted<Rhs, Spin>&& rhs) noexcept {
+    data_ += std::move(rhs.data());
```

Comment:
This is just acting with the underlying inplace operator; for `SpinWeighted`, the wrapped vector type is obtained with `data()`. It may be that this choice of naming causes the impression that there's a pointer being used, but I'm not sure of a better method name. (let me know if that's an insufficient explanation...)

It's important to not just use implicit casting or the like because we want to flag any time in the code the spin-weight is ignored during a math or assign operation.

Reply:
I was more curious (but did a terrible job being clear) about the meaning of `a += std::move(b)`.  

It doesn't seem like there is an opportunity for the data's ownership to change in this expression, so what is the std::move doing? After this call, does `a` contain the value of `a+b` as expected? does `b` still have the value `b`?

@moxcodes and I chatted in person about this and we aren't sure what this is about. @nilsdeppe do you have insight?

Reply:
There is no ownership change here. I vaguely remember us doing something like this in some coordinate map code where we were able to allied an allocation. Here the only way that could be possible is if `rhs` is of a different size than `*this`, but then I'd say you should actually error. In summary, I'm not familiar with `+=` type operators ever taking r-value references since there's generally speaking no optimization to be made (unless you allow adding things of different sizes, but let's not do that :) )

Reply:
@nilsdeppe So are you in favor of simply omitting the rvalue versions of these? I've included them simply for complete consistency of interface, naively forwarding to blaze under the assumption that (as always) blaze could be doing something I don't expect.

Reply:
Not a bad assumption :) In this case it's not possible as far as I'm aware, so I would remove them :)